olive/engine/footprint.py

```diff
@@ -0,0 +1,178 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+
+import logging
+from collections import OrderedDict
+from datetime import datetime
+from pathlib import Path
+
+from olive.common.config_utils import ConfigBase, config_json_dumps, config_json_loads
+
+logger = logging.getLogger(__name__)
+
+
+class FootprintNodeMetric(ConfigBase):
+    value: dict
+    cmp_direction: dict = {}
+    is_goals_met: bool = False
+
+
+class FootprintNode(ConfigBase):
```

Comment:
Can I also have the `inference_config` for a specific Pass output model?

Reply:
You can call the function of get_model_inference_config with given model_id to get the inference_config which is included in model_config.